spec/mongoid/document_persistence_context_spec.rb

```diff
@@ -30,4 +30,68 @@
       Person.collection_name.should == :people
     end
   end
+
+  context 'when loaded with an overridden persistence context' do
+    let(:options) { { collection: 'extra_people' } }
+    let(:person) { Person.with(options) { Person.create username: 'zyg14' } }
+
+    # Mongoid 9+ default persistence behavior
+    context 'when Mongoid.legacy_persistence_context_behavior is false' do
+      config_override :legacy_persistence_context_behavior, false
+
+      it 'remembers its persistence context when created' do
+        expect(person.collection_name).to be == :extra_people
+      end
+
+      it 'remembers its context when queried specifically' do
+        person_by_id = Person.with(options) { Person.find(_id: person._id) }
+        expect(person_by_id.collection_name).to be == :extra_people
+      end
+
+      it 'remembers its context when queried generally' do
+        person # force the person to be created
+        person_generally = Person.with(options) { Person.all[0] }
+        expect(person_generally.collection_name).to be == :extra_people
+      end
+
+      it 'can be reloaded without specifying the context' do
+        expect { person.reload }.not_to raise_error
+        expect(person.collection_name).to be == :extra_people
+      end
+
+      it 'can be updated without specifying the context' do
+        person.update username: 'zyg15'
+        expect(Person.with(options) { Person.first.username }).to be == 'zyg15'
+      end
```

Comment:
Please add a test which covers the case of overriding the context when setting/reloading

```
      it 'allows further overriding context when persisting' do
        person.username = 'bob'
        Person.with(collection: 'other') { person.save! }
        expect { person.reload.username }.to eq 'zyg14'
      end

      it 'allows further overriding context when reloading' do
        expect { Person.with(collection: 'other') { person.reload } }.to raise_error(Mongoid::Errors::DocumentNotFound)
      end
```